### PR TITLE
koord-manager: mutating pod support custom plugin

### PR DESCRIPTION
**Pull request by ZiMengSheng**
### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/webhook/pod/mutating/mutating_function.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package mutating
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type mutatingPodFunc func(h *PodMutatingHandler, ctx context.Context, req admission.Request, pod *corev1.Pod) error
```
Comment (eahydra) [inline pkg/webhook/pod/mutating/mutating_function.go:10]: IMO, make the function type as plugin is more flexible.  We cann add TODO comments to make the refactor later or add issue to track the issue. 

Add add more UT please.

**Comment by ZiMengSheng** [comment]
TBR

**Review by eahydra — COMMENTED** [review]
/lgtm

**Pull request closed without merging**